### DISCUSSION TITLE
Recording fields and parsing errors when Json data guard is used

### DISCUSSION THREAD

**Discussion by SSeanin** [rwf2/Rocket#2227] [Questions]
By using `Form<Contextual<'_, T>>` some errors could be returned from `FromFormField` impl in types deriving `FromForm`. These errors could be handled in the route handler by using `context`.

Is there a similar feature when using `Json<T>` as a data guard?

**Comment by the10thWiz** [comment]
Your best option is probably `Result<Json<T>, Error>`, which will report the errors in json parsing if it couldn't be parsed.

**Reply by SSeanin** [reply]
I actually ended up manually collecting the errors in a from conversion and forwarding them in the handler. This triggers another conversion between error types which could be handled. Different errors could be returned from handlers this way.
